spec/friendly_uuid_spec.rb

```diff
@@ -1,123 +1,147 @@
     end
 
     it 'returns an array when passed multiple IDs' do
-      found_item = Item.find(item_1.id, item_2.id)
+      found_items = Item.find(item_1.id, item_2.id)
 
-      expect(found_item).to be_kind_of Array
-      expect(found_item.length).to eq 2
+      expect(found_items.length).to eq 2
+      expect(found_items).to match_array([item_1, item_2])
     end
 
     it 'returns an array when passed multiple shortened IDs as an array' do
-      found_item = Item.find([Item.compact(item_1.id), Item.compact(item_2.id)])
+      found_items = Item.find([Item.compact(item_1.id), Item.compact(item_2.id)])
 
-      expect(found_item).to be_kind_of Array
-      expect(found_item.length).to eq 2
+      expect(found_items.length).to eq 2
+      expect(found_items).to match_array([item_1, item_2])
     end
 
     it 'returns an array when passed multiple IDs as an array' do
-      found_item = Item.find([item_1.id, item_2.id])
+      found_items = Item.find([item_1.id, item_2.id])
+
+      expect(found_items.length).to eq 2
+      expect(found_items).to match_array([item_1, item_2])
+    end
+
+    it 'raises ActiveRecord::RecordNotFound when passed nil' do
+      expect do
+        Item.find(nil)
+      end.to raise_error(ActiveRecord::RecordNotFound)
+    end
 
-      expect(found_item).to be_kind_of Array
-      expect(found_item.length).to eq 2
+    it 'raises ActiveRecord::RecordNotFound when passed an array of nil' do
+      expect do
+        Item.find([nil])
+      end.to raise_error(ActiveRecord::RecordNotFound)
+    end
+
+    it 'raises ActiveRecord::RecordNotFound when passed an empty array' do
+      expect do
+        Item.find([])
+      end.to raise_error(ActiveRecord::RecordNotFound)
     end
   end
 
-  describe '#find_by' do
+  describe '.find_by' do
     it 'returns an object when passed a shortened ID' do
       found_item = Item.find_by(id: Item.compact(item_1.id))
 
-      expect(found_item).to be_kind_of Item
+      expect(found_item).to eq item_1
     end
 
     it 'returns an object when passed an ID' do
       found_item = Item.find_by(id: item_1.id)
 
-      expect(found_item).to be_kind_of Item
+      expect(found_item).to eq item_1
     end
 
     it 'returns an object when passed a single shortened ID as an array' do
       found_item = Item.find_by(id: [Item.compact(item_1.id)])
 
-      expect(found_item).to be_kind_of Item
+      expect(found_item).to eq item_1
     end
 
     it 'returns an object when passed a single ID as an array' do
       found_item = Item.find_by(id: [item_1.id])
 
-      expect(found_item).to be_kind_of Item
+      expect(found_item).to eq item_1
     end
 
     it 'returns an object when passed multiple shortened IDs as an array' do
       found_item = Item.find_by(id: [Item.compact(item_1.id), Item.compact(item_2.id)])
 
-      expect(found_item).to be_kind_of Item
       expect(Array(found_item).length).to eq 1
+      expect(found_item).to be_kind_of Item
```

Comment:
Interestingly, changing to `eq item_1` results in this test intermittently failing since `.find_by` sometimes returns the first element in the array (`item_1`) and sometimes returns the second element (`item_2`). Possibly `ActiveRecord.find_by` performs similarly, but I haven't dug into it. Happy to do so, if you'd like, however!